src/test/java/com/puppycrawl/tools/checkstyle/api/FilterSetTest.java

```diff
@@ -117,6 +118,17 @@ public void testUnmodifiableSet() {
             () -> filterSet.getFilters().add(filter));
     }
 
+    /*
+      Due to low level configuration setup of FilterSet, conventional
+      input validation cannot be done here hence, pure JUnit testing has been
+      done for the time being
+    */
+    @Test
+    public void testEmptyToString() {
+        final FilterSet filterSet = new FilterSet();
+        Assertions.assertNotEquals(filterSet.toString(), "");
```

Comment:
please use Truth assertion api.

Reply:
I thought we already migrated all to Truth and forbid usage of jupiter assert api.

Reply:
@romani 
This has led to PMD violation saying the following:
```shell
[INFO] PMD version: 6.54.0
[INFO] PMD Failure: com.puppycrawl.tools.checkstyle.api.FilterSetTest:127 Rule:JUnitTestsShouldIncludeAssert Priority:3 JUnit tests should include assert() or fail().
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-pmd-plugin:3.20.0:check (default) on project checkstyle: You have 1 PMD violation. For more details see: C:\Users\shubh\OneDrive\Desktop\checkstyle project\checkstyle\target\pmd.xml -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
PS C:\Users\shubh\OneDrive\Desktop\checkstyle project\checkstyle>
```

Reply:
investigate why it is ok in all other cases but not in your code.

Reply:
@romani 

Please refer https://github.com/checkstyle/checkstyle/pull/12883#discussion_r1146533732 for why PMD violations occurred.

Reply:
@romani this problem has been fixed.